okta/resource_okta_app_oauth.go

```diff
@@ -254,6 +254,7 @@ func resourceAppOAuth() *schema.Resource {
 			"refresh_token_rotation": {
 				Type:             schema.TypeString,
 				Optional:         true,
+				Computed:         true,
```

Comment:
It's better to set `Default: "STATIC"`, since you can set `refresh_token_rotation` to `"ROTATE"` (however, this is an Early Access Feature)
```
resource "okta_app_oauth" "test" {
  label                  = "test"
  status                 = "ACTIVE"
  type                   = "browser"
  grant_types            = ["authorization_code", "refresh_token"]
  redirect_uris          = ["http://d.com/aaa"]
  response_types         = ["code"]
  hide_ios               = true
  hide_web               = true
  auto_submit_toolbar    = false
  refresh_token_rotation = "ROTATE"
  refresh_token_leeway   = 30
}
```

Reply:
Is it? It's certainly an option, but the Okta docs say it's only the default for certain app types, while SPAs (browser?) default to `ROTATE`. Terraform's `Default` field cannot accommodate any sort of conditional logic. 

Happy to change this but it seems like this would be a potential breaking change: users with apps where the server assigned `ROTATE` would see those resources changed to `STATIC`.

Reply:
You can try to create these two test apps via API (one with ROTATE, and one with STATIC)
```
{
    "name": "oidc_client",
    "label": "test",
    "status": "ACTIVE",
    "visibility": {
        "hide": {
            "iOS": true,
            "web": true
        },
        "appLinks": {
            "oidc_client_link": true
        }
    },
    "signOnMode": "OPENID_CONNECT",
    "settings": {
        "app": {},
        "oauthClient": {
            "redirect_uris": [
                "http://d.com/aaa"
            ],
            "response_types": [
                "code"
            ],
            "grant_types": [
                "authorization_code",
                "refresh_token"
            ],
            "application_type": "browser",
            "consent_method": "TRUSTED",
            "issuer_mode": "ORG_URL",
            "refresh_token": {
                "rotation_type": "STATIC"
            },
            "idp_initiated_login": {
                "mode": "DISABLED",
                "default_scope": []
            },
            "wildcard_redirect": "DISABLED"
        }
    }
}
```
And
```
{
    "name": "oidc_client",
    "label": "test",
    "status": "ACTIVE",
    "visibility": {
        "hide": {
            "iOS": true,
            "web": true
        },
        "appLinks": {
            "oidc_client_link": true
        }
    },
    "signOnMode": "OPENID_CONNECT",
    "settings": {
        "app": {},
        "oauthClient": {
            "redirect_uris": [
                "http://d.com/aaa"
            ],
            "response_types": [
                "code"
            ],
            "grant_types": [
                "authorization_code",
                "refresh_token"
            ],
            "application_type": "browser",
            "consent_method": "TRUSTED",
            "issuer_mode": "ORG_URL",
            "refresh_token": {
                "rotation_type": "ROTATE",
                "leeway": 30
            },
            "idp_initiated_login": {
                "mode": "DISABLED",
                "default_scope": []
            },
            "wildcard_redirect": "DISABLED"
        }
    }
}
```

Reply:
Thanks for those, the test case I have in mind is this:

```json
{
    "name": "oidc_client",
    "label": "test",
    "status": "ACTIVE",
    "visibility": {
        "hide": {
            "iOS": true,
            "web": true
        },
        "appLinks": {
            "oidc_client_link": true
        }
    },
    "signOnMode": "OPENID_CONNECT",
    "settings": {
        "app": {},
        "oauthClient": {
            "redirect_uris": [
                "http://d.com/aaa"
            ],
            "response_types": [
                "code"
            ],
            "grant_types": [
                "authorization_code",
                "refresh_token"
            ],
            "application_type": "browser",
            "consent_method": "TRUSTED",
            "issuer_mode": "ORG_URL",
            "idp_initiated_login": {
                "mode": "DISABLED",
                "default_scope": []
            },
            "wildcard_redirect": "DISABLED"
        }
    }
}
```

Essentially, your first case but with the `refresh_token` unset. The response I get, abbreviated for clarity:

```json
{
  "settings": {
    "oauthClient": {
      "application_type": "browser",
      "client_uri": null,
      "consent_method": "TRUSTED",
      "grant_types": [
        "authorization_code",
        "refresh_token"
      ],
      "idp_initiated_login": {
        "default_scope": [],
        "mode": "DISABLED"
      },
      "issuer_mode": "ORG_URL",
      "logo_uri": null,
      "redirect_uris": [
        "http://d.com/aaa"
      ],
      "refresh_token": {
        "leeway": 30,
        "rotation_type": "ROTATE"
      },
      "response_types": [
        "code"
      ]
    }
  }
}
```

Note that `settings.refresh_token.rotation_type` is `"ROTATE"`.

Given the following configuration:

```tf
resource "okta_app_oauth" "app" {
  type = "browser"
}
```

Currently, that should result in `refresh_token` being set to `"ROTATE"` in state, since the server returns a default. There will be a perpetual diff given that the user-specified value is `null`. If the `Default` were changed, the next apply would change the setting to `"STATIC"` for the live app. 

The change I've proposed here should eliminate the perpetual diff without causing changes to existing resources. Let me know if I'm missing something though.

Reply:
Hi, just want to re-raise this, still seems like marketing this key as computed is the appropriate fix